### PR TITLE
Aden 3531 interstitial test fix

### PR DESCRIPTION
**Pull request by jacekjursza**
* Wait for close button
* Wait for interstitial itself (waiting for page load is not always enough)

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java [hunk 1]
```diff
@@ -30,17 +33,30 @@
   @FindBy(css = "iframe.wikia-ad-iframe")
   private WebElement interstitialAdIframe;
 
-  @FindBy(css = "div#ext-wikia-adEngine-template-modal, div.lightbox-content-inner > div")
+  static final String interstitialAdWrapperSelector = "div#ext-wikia-adEngine-template-modal, div.lightbox-content-inner > div";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java]: MAJOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00115)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java [hunk 2]
```diff
@@ -77,7 +93,23 @@ public void verifySize(Dimension expectedSize) {
     );
   }
 
-  public void closeInterstitial() {
+  public void closeInterstitial() throws InterruptedException {
+
+    boolean isMercury = false;
+    try {
+      isMercury = interstitialCloseButtonMercury.isDisplayed();
+    } catch (NoSuchElementException e) { /* ignore */ }
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsInterstitial.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package com.wikia.webdriver.testcases.adstests;
 
 import com.wikia.webdriver.common.core.annotations.RelatedIssue;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsInterstitial.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.annotations.RelatedIssue'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsInterstitial.java [hunk 4]
```diff
@@ -1,6 +1,7 @@
...
+import com.wikia.webdriver.common.core.elemnt.JavascriptActions;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsInterstitial.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java [hunk 5]
```diff
@@ -77,7 +95,26 @@ public void verifySize(Dimension expectedSize) {
...
+      PageObjectLogging.log("Interstitial Mercury", "No", true);
+    }
+
+
+    if (isMercury) {
+      String closeButtonText = interstitialCloseButtonMercury.getText();
+      if (closeButtonText.length() > 0) {
+        Integer waitTillCloseButtonAppears = Integer.parseInt(closeButtonText);
+        PageObjectLogging.log("Wait time for close button", String.valueOf(waitTillCloseButtonAppears), true);
+        Thread.sleep((waitTillCloseButtonAppears+2) * 1000);
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java]: Where does `2` come from? ;-)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java [hunk 6]
```diff
@@ -30,17 +33,32 @@
...
+  private WebElement interstitialCounterOasis;
+
   public AdsInterstitialObject(WebDriver driver, String testedPage,
                                Dimension resolution) {
     super(driver, testedPage, resolution);
   }
 
+  public void waitForInterstitialShowUp() {
+    JavascriptActions javascriptActions = new JavascriptActions(driver);
+    javascriptActions.waitForJavaScriptTruthy("document.querySelectorAll(\"" + INTERSTITIAL_AD_WRAPPER_SELECTOR
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java]: Minor thing: how about using `String.format()` and use `'` within `querySelectorAll()` to make it more readable with fewer slashes? ;)

**Comment by nandy-andy** [comment]
Looks promising :+1:

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java [hunk 7]
```diff
@@ -77,7 +98,26 @@ public void verifySize(Dimension expectedSize) {
...
   }
 
-  public void closeInterstitial() {
+  public void closeInterstitial() throws InterruptedException {
+
+    boolean isMercury = false;
+    try {
+      isMercury = interstitialCloseButtonMercury.isDisplayed();
+      PageObjectLogging.log("Interstitial Mercury", "Yes", true);
+    } catch (NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsInterstitialObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by nandy-andy** [comment]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**